yt/geometry/selection_routines.pyx

```diff
@@ -1473,10 +1473,19 @@ cdef class SliceSelector(SelectorObject):
     cdef int axis
     cdef np.float64_t coord
     cdef int ax, ay
+    cdef int reduced_dimensionality
 
     def __init__(self, dobj):
         self.axis = dobj.axis
         self.coord = _ensure_code(dobj.coord)
+        # If we have a reduced dimensionality dataset, we want to avoid any
+        # checks against it in the axes that are beyond its dimensionality.
+        # This means that if we have a 2D dataset, *all* slices along z will
+        # select all the zones.
+        if self.axis >= dobj.ds.dimensionality:
+            self.reduced_dimensionality = 1
```

Comment:
So this attribute is essentially a boolean, right ? Any reason not to use True and False as values ? (maybe that's not recommended in Cython)

Reply:
Ah!  So `bool` doesn't work as a cdef.  I think `bint` might but I was hoping to keep it as close to C and not C++ as possible.

Reply:
alright that works for me !